### ISSUE TITLE
Proposal: Unify Sound and Music event

### ISSUE THREAD

**Issue by Jowan-Spooner** [dialogic-godot/dialogic#1306]
These events are currently VERY similar except:
- Music has a loop and fade option
- They use different audio players

It could make sense to make it one event. However this is just an idea. Would like to hear your thoughts. Thought of this as it's how it looks in @coppolaemilio's mockup (that is used in the readme).

**Comment by thebardsrc** [comment]
one event would probly be fine, but they would still need two separate buses/channels/whatever (idk not familiar with how Godot audio works), for the purpose of players being able to adjust the volume and sound effect levels independently

**Comment by zaknafean** [comment]
@KvaGram is probably a good one to have in this discussion.

**Comment by Jowan-Spooner** [comment]
Well I would still have two modes (music and sound)  on the event. Audio bus can be set by the user anyways, but maybe adding a default (changeable in the settings) for both modes would be good as well.

**Comment by KvaGram** [comment]
KvaGram the Autistic Duck reporting in.


> one event would probly be fine, but they would still need two separate buses/channels/whatever (idk not familiar with how Godot audio works), for the purpose of players being able to adjust the volume and sound effect levels independently
> 

The problem is with Dialogic. As @Jowan-Spooner pointed out, each even have their own player.
A player can only play one soundfile at a time. So if you want music and sound effects at the same time, there must be separate players. Same goes with voiceovers and type-sound effects. That's 4 players in all, unless I'm missing something.

If we were to unify the events, we would need to be able to define what player to use in the event. That requires aditional datafield in the UI of the event.
A common complaint of some of the events, is the clutter.

We could have dialogic just have a pool of players, using them one by one. That would solve the problem of having to set one. But that would introduce a new problem. What if you wanted to stop one specific track?
There are fixes to that too, but that again would mean adding another datafield to an event.
There is no way around that.

> For the voice overs too, for that matter, thafs one people would want to control as well. Three separate things with independent volume

I am dreaming of means to better control voice overs. I am kinda out of it right now. But when I get my shit together and start coding again, I got some ideas I itch to try out.

> Well I would still have two modes (music and sound) on the event. Audio bus can be set by the user anyways, but maybe adding a default (changeable in the settings) for both modes would be good as well.

The audiobus don't matter in this case. We already have an option for the audiobus, well it's more like a compulsory field that defaults to the master-bus. It already has to be set. Again, the limit is in dialogic's audio players.

**Comment by KvaGram** [comment]
Here is my proposal to fix this.
Add an index-field to the unified sound/music field ( integer [0, 100] defaults to 0)
Include a loop-checkbox (boolean, defaults to off)

The audio subsystem checks if an audioplayer given that index exists. If it does, it will stop any sound playing.
If the index does not exist yet, a new audioplayer is created and assigned to that index.
Next the the new audiofile and settings are set and it plays.
If loop is enabled, the audio subsystem's code will ensure the track loops.

That said, I am unsure if we should do this. But this is how I would do it.

As for also merging voice over, I think that is a very bad idea. It requires additional datafields not used in neither sound effects nor music.

**Comment by thebardsrc** [comment]
> As for also merging voice over, I think that is a very bad idea. It requires additional datafields not used in neither sound effects nor music.

Yes of course, I wasn't suggesting that. That was just a tangential of that all needs it's own separate volume control too, like I said I don't really understand Godot's audio system in general much less how Dialogic has it hooked up and whether those already have their own volume or are controlled by the master volume

**Comment by salianifo** [comment]
I don't feel this is needed now that we have multiple music channels, it feels like music and sounds are sufficiently different to warrant separate events. Additionally, although they both are audio things with similar settings, semantically they represent different things, and should be left separate in my opinion.
